azurelinuxagent/agent.py

```diff
@@ -199,6 +202,26 @@ def collect_logs(self, is_full_mode):
         else:
             print("Running log collector mode normal")
 
+        # Check the cgroups unit
+        if CollectLogsHandler.should_validate_cgroups():
+            cgroup_path_regex = re.compile(r'^(?P<slice>.*)/(?P<unit>.*)$')
+
+            cpu_cgroup_path, memory_cgroup_path = SystemdCgroupsApi.get_process_cgroup_relative_paths("self")
```

Comment:
Note that get_process_cgroup_relative_paths can return None if, for example, the CPU and/or memory controllers are not enabled. You need to protect your code against this condition.

Reply:
the newest commit should handle this case. Let me know! thanks.